src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java

```diff
@@ -68,17 +68,32 @@ public LocalUpdateSyncRequest(String thingName,
      * @throws IOException if unable to serialize the update document payload bytes.
      */
     public void merge(LocalUpdateSyncRequest other) throws IOException {
-        Optional<JsonNode> oldValueJson = JsonUtil.getPayloadJson(updateDocument);
-        Optional<JsonNode> newValueJson = JsonUtil.getPayloadJson(other.getUpdateDocument());
-        if (!oldValueJson.isPresent() && newValueJson.isPresent()) {
+        Optional<JsonNode> currDocJson = JsonUtil.getPayloadJson(updateDocument);
+        Optional<JsonNode> updateDocJson = JsonUtil.getPayloadJson(other.getUpdateDocument());
+
+        if (!currDocJson.isPresent() && updateDocJson.isPresent()) {
             updateDocument = other.getUpdateDocument();
             return;
         }
-        if (!newValueJson.isPresent()) {
+        if (!updateDocJson.isPresent()) {
             return;
         }
-        JsonMerger.merge(oldValueJson.get(), newValueJson.get());
-        updateDocument = JsonUtil.getPayloadBytes(oldValueJson.get());
+
+        JsonNode oldValueJson = currDocJson.get();
+        JsonNode newValueJson = updateDocJson.get();
+
+        if (JsonUtil.hasVersion(oldValueJson) && JsonUtil.hasVersion(newValueJson)
```

Comment:
I feel this is a good start. But we might want to consider a per node update instead of a per version update in the future. It is a bit more work. But that will ensure that the data is correct.